packages/expo-cli/src/credentials/views/__tests__/AndroidKeystore-Remove-test.ts

```diff
@@ -0,0 +1,105 @@
+import prompt from '../../../prompt';
+import { mockExpoXDL } from '../../../__tests__/mock-utils';
+
+jest.mock('../../actions/list');
+jest.mock('../../../prompt');
+jest.mock('fs-extra');
+mockExpoXDL({
+  AndroidCredentials: {
+    generateUploadKeystore: jest.fn().mockImplementation(() => ({
+      keystorePth: 'test',
+      keystorePassword: 'test',
+      keyAlias: 'test',
+      keyPassword: 'test',
+    })),
+  },
+});
+
+const originalWarn = console.warn;
+const originalLog = console.log;
+beforeAll(() => {
+  console.warn = jest.fn();
+  console.log = jest.fn();
+});
+afterAll(() => {
+  console.warn = originalWarn;
+  console.log = originalLog;
+});
+beforeEach(() => {
+  (prompt as any).mockReset();
+  (prompt as any).mockImplementation(() => {
+    throw new Error('Should not be called');
+  });
+});
+
+describe('RemoveKeystore', () => {
+  describe('existing credentials', () => {
+    it('should display warning prompt and abort', async () => {
+      const ctx = getCtxMock();
+      (prompt as any)
+        .mockImplementationOnce(() => ({ confirm: false })) // prompt with warning message, abort
+        .mockImplementationOnce(x => {
+          throw new Error("shouldn't happen");
+        });
+
+      const view = new RemoveKeystore(testExperienceName);
+      const lastView = await view.open(ctx);
+
+      expect(lastView).toBe(null);
+      expect((prompt as any).mock.calls.length).toBe(1);
+      expect(ctx.android.fetchKeystore.mock.calls.length).toBe(1);
+      expect(ctx.android.removeKeystore.mock.calls.length).toBe(0);
+    });
+
+    it('should display warning prompt and continue', async () => {
+      const ctx = getCtxMock();
+      fs.pathExists.mockImplementationOnce(() => false);
+
+      // first: prompt with warning message, true means continue
+      // second: ask if cli should display credentials, user won't see that because when() should return false
+      (prompt as any)
+        .mockImplementationOnce(() => ({ confirm: true }))
+        .mockImplementationOnce(question => {
+          if (question.when()) {
+            throw new Error("shouldn't happen");
+          }
+          return { confirm: undefined };
+        })
+        .mockImplementationOnce(() => {
+          throw new Error("shouldn't happen");
+        });
+
+      const view = new RemoveKeystore(testExperienceName);
+      const lastView = await view.open(ctx);
+
+      expect(lastView).toBe(null);
+      expect((prompt as any).mock.calls.length).toBe(2);
+      expect(ctx.android.fetchKeystore.mock.calls.length).toBe(2);
+      expect(ctx.android.removeKeystore.mock.calls.length).toBe(1);
+    });
+  });
+
+  describe('no credentials', () => {
+    it("shouldn't display warning prompt and finish", async () => {
+      const ctx = getCtxMock({
+        android: {
+          fetchKeystore: jest.fn().mockImplementationOnce(() => null),
+        },
+      });
+      (prompt as any).mockImplementationOnce(() => {
```

Comment:
```suggestion
      (prompt as any).mockImplementation(() => {
```

Reply:
Even though they have the same end behaviour in this case, I'd prefer `mockImplementation` because it signals to the reader that it is the default behaviour we want, instead of some specific behaviour we are looking for on the nth call of the method.